packages/@angular/cli/utilities/validate-project-name.ts

```diff
@@ -2,30 +2,33 @@ import {oneLine, stripIndent} from 'common-tags';
 
 const SilentError = require('silent-error');
 
-const projectNameRegexp = /^[a-zA-Z][.0-9a-zA-Z]*(-[.0-9a-zA-Z]*)*$/;
+const projectNameRegexp = /^[a-zA-Z][0-9a-zA-Z]*([-. ][a-zA-Z][0-9a-zA-Z]*)*$/;
```

Comment:
Yes @hansl this PR specifically validates all these 3 points.
FYI. for point#3. since I have looked into the angular-cli code it replaces all spaces in project-name with dashes for valid npm-package-name when writing package.json file.